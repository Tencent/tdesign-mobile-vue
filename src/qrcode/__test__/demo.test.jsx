/**
 * 该文件为由脚本 `npm run test:demo` 自动生成，如需修改，执行脚本命令即可。请勿手写直接修改，否则会被覆盖
 */

import { mount } from '@vue/test-utils';
import baseVue from '@/qrcode/demos/base.vue';
import colorVue from '@/qrcode/demos/color.vue';
import downloadVue from '@/qrcode/demos/download.vue';
import iconVue from '@/qrcode/demos/icon.vue';
import levelVue from '@/qrcode/demos/level.vue';
import mobileVue from '@/qrcode/demos/mobile.vue';
import sizeVue from '@/qrcode/demos/size.vue';
import statusRenderVue from '@/qrcode/demos/status-render.vue';
import statusVue from '@/qrcode/demos/status.vue';
import typeVue from '@/qrcode/demos/type.vue';

const mapper = {
  baseVue,
  colorVue,
  downloadVue,
  iconVue,
  levelVue,
  mobileVue,
  sizeVue,
  statusRenderVue,
  statusVue,
  typeVue,
};

describe('Qrcode', () => {
  Object.keys(mapper).forEach((demoName) => {
    it(`Qrcode ${demoName} demo works fine`, () => {
      const wrapper = mount(mapper[demoName]);
      expect(wrapper.element).toMatchSnapshot();
    });
  });
});
