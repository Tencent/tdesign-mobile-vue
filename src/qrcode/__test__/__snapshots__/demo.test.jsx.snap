// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Qrcode > Qrcode baseVue demo works fine 1`] = `
<div>
  <div
    class="t-qrcode"
    level="M"
    style="background-color: transparent; width: 160px; height: 160px;"
  >
    <!---->
    
    <canvas
      role="img"
    />
    <!---->
    
  </div>
</div>
`;

exports[`Qrcode > Qrcode borderlessVue demo works fine 1`] = `
<div>
  <div
    class="t-qrcode t-borderless"
    level="M"
    style="background-color: transparent; width: 160px; height: 160px;"
  >
    <!---->
    
    <canvas
      role="img"
    />
    <!---->
    
  </div>
</div>
`;

exports[`Qrcode > Qrcode colorVue demo works fine 1`] = `
<div
  style="display: flex; flex-direction: column;"
>
  <div
    class="t-qrcode"
    level="M"
    style="background-color: rgb(255, 255, 255); width: 160px; height: 160px;"
  >
    <!---->
    
    <canvas
      role="img"
    />
    <!---->
    
  </div>
  <p
    class="tdesign-mobile-demo-block__summary"
  >
    二维码背景颜色
  </p>
  <div
    class="t-qrcode"
    level="M"
    style="background-color: rgb(217, 225, 255); width: 160px; height: 160px;"
  >
    <!---->
    
    <canvas
      role="img"
    />
    <!---->
    
  </div>
</div>
`;

exports[`Qrcode > Qrcode downloadVue demo works fine 1`] = `
<div
  data-v-848000e9=""
  style="display: flex; flex-direction: column;"
>
  <div
    class="button-group"
    data-v-848000e9=""
  >
    
    <button
      aria-disabled="false"
      class="t-button t-button--size-medium t-button--base t-button--light t-button--round"
      data-v-848000e9=""
      role="button"
      type="button"
    >
      <!---->
      <span
        class="t-button__content"
      >
        
        canvas
        
      </span>
      <!---->
    </button>
    <button
      aria-disabled="false"
      class="t-button t-button--size-medium t-button--base t-button--default t-button--round"
      data-v-848000e9=""
      role="button"
      type="button"
    >
      <!---->
      <span
        class="t-button__content"
      >
        
        svg
        
      </span>
      <!---->
    </button>
    
  </div>
  <div
    class="t-qrcode"
    data-v-848000e9=""
    id="QRCode"
    level="M"
    style="background-color: transparent; width: 160px; height: 160px;"
  >
    <!---->
    
    <canvas
      role="img"
    />
    <img
      src="https://tdesign.gtimg.com/site/tdesign-logo.png"
      style="display: none;"
    />
    
  </div>
  <div
    class="button-group"
    data-v-848000e9=""
  >
    <button
      aria-disabled="false"
      class="t-button t-button--size-medium t-button--base t-button--primary t-button--rectangle"
      data-v-848000e9=""
      role="button"
      type="button"
    >
      <!---->
      <span
        class="t-button__content"
      >
        
        Download
        
      </span>
      <!---->
    </button>
  </div>
</div>
`;

exports[`Qrcode > Qrcode iconVue demo works fine 1`] = `
<div>
  <div
    class="t-qrcode"
    level="M"
    style="background-color: transparent; width: 160px; height: 160px;"
  >
    <!---->
    
    <canvas
      role="img"
    />
    <img
      src="https://tdesign.gtimg.com/site/tdesign-logo.png"
      style="display: none;"
    />
    
  </div>
</div>
`;

exports[`Qrcode > Qrcode levelVue demo works fine 1`] = `
<div
  data-v-app=""
>
  
  <div
    class="t-slider"
  >
    <!---->
    <div
      class="t-slider__bar t-slider__bar--default"
    >
      <!---->
      <div
        class="t-slider__line t-slider__line--default t-slider__line--single"
        style=""
      >
        <div
          class="t-slider__dot"
        >
          <!---->
          <div
            class="t-slider__dot-slider"
          />
        </div>
      </div>
    </div>
    <!---->
  </div>
  <div
    class="qrcode-content"
  >
    <div
      class="t-qrcode"
      level="M"
      style="background-color: transparent; width: 160px; height: 160px;"
    >
      <!---->
      
      <canvas
        role="img"
      />
      <!---->
      
    </div>
  </div>
  
</div>
`;

exports[`Qrcode > Qrcode mobileVue demo works fine 1`] = `
<div
  class="tdesign-mobile-demo"
>
  <h1
    class="title"
  >
    QRCode 二维码
  </h1>
  <p
    class="summary"
  >
    能够将文本转换生成二维码的组件，支持自定义配色和 Logo 配置。
  </p>
  <div
    class="tdesign-mobile-demo-block"
  >
    <div
      class="tdesign-mobile-demo-block__header"
    >
      <h2
        class="tdesign-mobile-demo-block__title"
      >
        01 组件类型
      </h2>
      <p
        class="tdesign-mobile-demo-block__summary"
      >
        基本用法
      </p>
    </div>
    <div
      class="tdesign-mobile-demo-block__slot with-padding"
    >
      
      <div>
        <div
          class="t-qrcode"
          level="M"
          style="background-color: transparent; width: 160px; height: 160px;"
        >
          <!---->
          
          <canvas
            role="img"
          />
          <!---->
          
        </div>
      </div>
      
    </div>
  </div>
  <div
    class="tdesign-mobile-demo-block tdesign-mobile-demo-block_notitle"
  >
    <div
      class="tdesign-mobile-demo-block__header"
    >
      <!--v-if-->
      <p
        class="tdesign-mobile-demo-block__summary tdesign-mobile-demo-block_subtitle"
      >
        带 Icon 的二维码
      </p>
    </div>
    <div
      class="tdesign-mobile-demo-block__slot with-padding"
    >
      
      <div>
        <div
          class="t-qrcode"
          level="M"
          style="background-color: transparent; width: 160px; height: 160px;"
        >
          <!---->
          
          <canvas
            role="img"
          />
          <img
            src="https://tdesign.gtimg.com/site/tdesign-logo.png"
            style="display: none;"
          />
          
        </div>
      </div>
      
    </div>
  </div>
  <div
    class="tdesign-mobile-demo-block tdesign-mobile-demo-block_notitle"
  >
    <div
      class="tdesign-mobile-demo-block__header"
    >
      <!--v-if-->
      <p
        class="tdesign-mobile-demo-block__summary tdesign-mobile-demo-block_subtitle"
      >
        无边框的二维码
      </p>
    </div>
    <div
      class="tdesign-mobile-demo-block__slot with-padding"
    >
      
      <div>
        <div
          class="t-qrcode t-borderless"
          level="M"
          style="background-color: transparent; width: 160px; height: 160px;"
        >
          <!---->
          
          <canvas
            role="img"
          />
          <!---->
          
        </div>
      </div>
      
    </div>
  </div>
  <div
    class="tdesign-mobile-demo-block tdesign-mobile-demo-block_notitle"
  >
    <div
      class="tdesign-mobile-demo-block__header"
    >
      <!--v-if-->
      <p
        class="tdesign-mobile-demo-block__summary tdesign-mobile-demo-block_subtitle"
      >
        二维码纠错等级
      </p>
    </div>
    <div
      class="tdesign-mobile-demo-block__slot"
    >
      
      
      <div
        class="t-slider"
      >
        <!---->
        <div
          class="t-slider__bar t-slider__bar--default"
        >
          <!---->
          <div
            class="t-slider__line t-slider__line--default t-slider__line--single"
            style=""
          >
            <div
              class="t-slider__dot"
            >
              <!---->
              <div
                class="t-slider__dot-slider"
              />
            </div>
          </div>
        </div>
        <!---->
      </div>
      <div
        class="qrcode-content"
      >
        <div
          class="t-qrcode"
          level="M"
          style="background-color: transparent; width: 160px; height: 160px;"
        >
          <!---->
          
          <canvas
            role="img"
          />
          <!---->
          
        </div>
      </div>
      
      
    </div>
  </div>
  <!-- &lt;tdesign-demo-block summary="自定义渲染类型"&gt;
      &lt;TypeDemo  /&gt;
    &lt;/tdesign-demo-block&gt; -->
  <div
    class="tdesign-mobile-demo-block"
  >
    <div
      class="tdesign-mobile-demo-block__header"
    >
      <h2
        class="tdesign-mobile-demo-block__title"
      >
        02 组件状态
      </h2>
      <p
        class="tdesign-mobile-demo-block__summary"
      >
        active
      </p>
    </div>
    <div
      class="tdesign-mobile-demo-block__slot"
    >
      
      <div
        class="tdesign-demos-qrcode"
        style="display: flex; flex-direction: column;"
      >
        <div
          class="t-qrcode"
          level="M"
          style="background-color: transparent; width: 160px; height: 160px;"
        >
          <!---->
          
          <canvas
            role="img"
          />
          <!---->
          
        </div>
        <p
          class="tdesign-mobile-demo-block__summary"
        >
          expired
        </p>
        <div
          class="t-qrcode"
          level="M"
          style="background-color: transparent; width: 160px; height: 160px;"
        >
          <div
            class="t-mask t-expired"
          >
            
            
            <p
              class="t-expired__text"
            >
              二维码过期
            </p>
            <p
              class="t-expired__button"
            >
              <svg
                class="t-icon t-icon-refresh"
                fill="none"
                height="1em"
                viewBox="0 0 24 24"
                width="1em"
              >
                <path
                  d="M20.4997 5.83468C18.5918 3.20865 15.4961 1.5 11.9997 1.5C6.57341 1.5 2.10958 5.61542 1.55717 10.896L1.45312 11.8905L3.44227 12.0986L3.54632 11.104C3.9933 6.83125 7.60814 3.5 11.9997 3.5C15.0404 3.5 17.7099 5.09688 19.2125 7.5H15.4997V9.5H22.4997V2.5H20.4997V5.83468ZM20.5572 11.9014L20.4532 12.896C20.0062 17.1687 16.3913 20.5 11.9997 20.5C8.95904 20.5 6.28956 18.9031 4.78702 16.5H8.49974V14.5H1.49974V21.5H3.49974V18.1653C5.40769 20.7914 8.50342 22.5 11.9997 22.5C17.4261 22.5 21.8899 18.3846 22.4423 13.104L22.5464 12.1095L20.5572 11.9014Z"
                  fill="currentColor"
                />
              </svg>
              点击刷新
            </p>
            
            
          </div>
          
          <canvas
            role="img"
          />
          <!---->
          
        </div>
        <p
          class="tdesign-mobile-demo-block__summary"
        >
          loading
        </p>
        <div
          class="t-qrcode"
          level="M"
          style="background-color: transparent; width: 160px; height: 160px;"
        >
          <div
            class="t-mask"
          >
            
            <div
              class="t-loading"
              style="font-size: 32px;"
            >
              <svg
                class="t-loading__gradient t-icon-loading"
                height="1em"
                style="animation-duration: 800ms; width: 32px; height: 32px;"
                version="1.1"
                viewBox="0 0 12 12"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <foreignobject
                  height="12"
                  width="12"
                  x="0"
                  y="0"
                >
                  <div
                    class="t-loading__gradient-conic"
                  />
                </foreignobject>
              </svg>
              <!---->
              <!---->
            </div>
            
          </div>
          
          <canvas
            role="img"
          />
          <!---->
          
        </div>
        <p
          class="tdesign-mobile-demo-block__summary"
        >
          scanned
        </p>
        <div
          class="t-qrcode"
          level="M"
          style="background-color: transparent; width: 160px; height: 160px;"
        >
          <div
            class="t-mask t-scanned"
          >
            
            <p
              class="t-scanned"
            >
              <svg
                class="t-icon t-icon-check-circle-filled t-scanned__icon"
                fill="none"
                height="1em"
                viewBox="0 0 24 24"
                width="1em"
              >
                <path
                  d="M12 23C18.0751 23 23 18.0751 23 12C23 5.92487 18.0751 1 12 1C5.92487 1 1 5.92487 1 12C1 18.0751 5.92487 23 12 23ZM7.49985 10.5858L10.4999 13.5858L16.4999 7.58578L17.9141 8.99999L10.4999 16.4142L6.08564 12L7.49985 10.5858Z"
                  fill="currentColor"
                />
              </svg>
              已扫描
            </p>
            
          </div>
          
          <canvas
            role="img"
          />
          <!---->
          
        </div>
      </div>
      
    </div>
  </div>
  <!-- &lt;tdesign-demo-block summary="自定义状态渲染器"&gt;
      &lt;StatusRenderDemo  /&gt;
    &lt;/tdesign-demo-block&gt; -->
  <div
    class="tdesign-mobile-demo-block"
  >
    <div
      class="tdesign-mobile-demo-block__header"
    >
      <h2
        class="tdesign-mobile-demo-block__title"
      >
        03 组件样式
      </h2>
      <p
        class="tdesign-mobile-demo-block__summary"
      >
        二维码颜色
      </p>
    </div>
    <div
      class="tdesign-mobile-demo-block__slot"
    >
      
      <div
        class="tdesign-demos-qrcode"
        style="display: flex; flex-direction: column;"
      >
        <div
          class="t-qrcode"
          level="M"
          style="background-color: rgb(255, 255, 255); width: 160px; height: 160px;"
        >
          <!---->
          
          <canvas
            role="img"
          />
          <!---->
          
        </div>
        <p
          class="tdesign-mobile-demo-block__summary"
        >
          二维码背景颜色
        </p>
        <div
          class="t-qrcode"
          level="M"
          style="background-color: rgb(217, 225, 255); width: 160px; height: 160px;"
        >
          <!---->
          
          <canvas
            role="img"
          />
          <!---->
          
        </div>
      </div>
      
    </div>
  </div>
  <div
    class="tdesign-mobile-demo-block tdesign-mobile-demo-block_notitle"
  >
    <div
      class="tdesign-mobile-demo-block__header"
    >
      <!--v-if-->
      <p
        class="tdesign-mobile-demo-block__summary tdesign-mobile-demo-block_subtitle"
      >
        二维码尺寸
      </p>
    </div>
    <div
      class="tdesign-mobile-demo-block__slot with-padding"
    >
      
      <div
        data-v-55577f76=""
        style="display: flex; flex-direction: column;"
      >
        <div
          class="button-group"
          data-v-55577f76=""
        >
          <button
            aria-disabled="false"
            class="t-button t-button--size-medium t-button--text t-button--default t-button--rectangle button"
            data-v-55577f76=""
            role="button"
            type="button"
          >
            <!---->
            <span
              class="t-button__content"
            >
              
              - Smaller
              
            </span>
            <!---->
          </button>
          <div
            class="line"
            data-v-55577f76=""
          />
          <button
            aria-disabled="false"
            class="t-button t-button--size-medium t-button--text t-button--default t-button--rectangle button"
            data-v-55577f76=""
            role="button"
            type="button"
          >
            <!---->
            <span
              class="t-button__content"
            >
              
              + Larger
              
            </span>
            <!---->
          </button>
        </div>
        <div
          class="container"
          data-v-55577f76=""
        >
          <div
            class="t-qrcode"
            data-v-55577f76=""
            level="M"
            style="background-color: transparent; width: 160px; height: 160px;"
          >
            <!---->
            
            <canvas
              role="img"
            />
            <!---->
            
          </div>
        </div>
      </div>
      
    </div>
  </div>
  <!-- &lt;tdesign-demo-block summary="下载二维码的简单实现。"&gt;
      &lt;DownloadDemo  /&gt;
    &lt;/tdesign-demo-block&gt; -->
</div>
`;

exports[`Qrcode > Qrcode sizeVue demo works fine 1`] = `
<div
  data-v-55577f76=""
  style="display: flex; flex-direction: column;"
>
  <div
    class="button-group"
    data-v-55577f76=""
  >
    <button
      aria-disabled="false"
      class="t-button t-button--size-medium t-button--text t-button--default t-button--rectangle button"
      data-v-55577f76=""
      role="button"
      type="button"
    >
      <!---->
      <span
        class="t-button__content"
      >
        
        - Smaller
        
      </span>
      <!---->
    </button>
    <div
      class="line"
      data-v-55577f76=""
    />
    <button
      aria-disabled="false"
      class="t-button t-button--size-medium t-button--text t-button--default t-button--rectangle button"
      data-v-55577f76=""
      role="button"
      type="button"
    >
      <!---->
      <span
        class="t-button__content"
      >
        
        + Larger
        
      </span>
      <!---->
    </button>
  </div>
  <div
    class="container"
    data-v-55577f76=""
  >
    <div
      class="t-qrcode"
      data-v-55577f76=""
      level="M"
      style="background-color: transparent; width: 160px; height: 160px;"
    >
      <!---->
      
      <canvas
        role="img"
      />
      <!---->
      
    </div>
  </div>
</div>
`;

exports[`Qrcode > Qrcode statusRenderVue demo works fine 1`] = `
<div>
  <!-- 自定义状态-使用Function -->
  <div
    class="t-qrcode"
    level="M"
    style="background-color: transparent; width: 160px; height: 160px;"
  >
    <div
      class="t-mask"
    >
      
      <div
        style="display: flex; flex-direction: column; align-items: center; gap: 4px;"
      >
        <div
          class="t-loading"
          style="font-size: 32px;"
        >
          <svg
            class="t-loading__gradient t-icon-loading"
            height="1em"
            style="animation-duration: 800ms; width: 32px; height: 32px;"
            version="1.1"
            viewBox="0 0 12 12"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <foreignobject
              height="12"
              width="12"
              x="0"
              y="0"
            >
              <div
                class="t-loading__gradient-conic"
              />
            </foreignobject>
          </svg>
          <!---->
          <!---->
        </div>
        <p>
          加载中...
        </p>
      </div>
      
    </div>
    
    <canvas
      role="img"
    />
    <!---->
    
  </div>
  <!-- 自定义状态-使用插槽 -->
  <div
    class="t-qrcode"
    level="M"
    style="background-color: transparent; width: 160px; height: 160px;"
  >
    <div
      class="t-mask t-expired"
    >
      
      
      <div
        class="tdesign-demo-qrcode__status-render"
      >
        <p
          class="tdesign-demo-qrcode__status-render-title"
        >
          <svg
            class="t-icon t-icon-close-circle-filled"
            fill="none"
            height="1em"
            viewBox="0 0 24 24"
            width="1em"
          >
            <path
              d="M12 23C18.0751 23 23 18.0751 23 12C23 5.92487 18.0751 1 12 1C5.92487 1 1 5.92487 1 12C1 18.0751 5.92487 23 12 23ZM8.81753 7.40346L11.9999 10.5858L15.1815 7.40414L16.5957 8.81835L13.4141 12L16.5957 15.1816L15.1815 16.5958L11.9999 13.4142L8.81753 16.5965L7.40332 15.1823L10.5856 12L7.40332 8.81767L8.81753 7.40346Z"
              fill="currentColor"
            />
          </svg>
          <span>
            二维码过期
          </span>
        </p>
        <p
          class="tdesign-demo-qrcode__status-render-action"
        >
          <svg
            class="t-icon t-icon-refresh"
            fill="none"
            height="1em"
            viewBox="0 0 24 24"
            width="1em"
          >
            <path
              d="M20.4997 5.83468C18.5918 3.20865 15.4961 1.5 11.9997 1.5C6.57341 1.5 2.10958 5.61542 1.55717 10.896L1.45312 11.8905L3.44227 12.0986L3.54632 11.104C3.9933 6.83125 7.60814 3.5 11.9997 3.5C15.0404 3.5 17.7099 5.09688 19.2125 7.5H15.4997V9.5H22.4997V2.5H20.4997V5.83468ZM20.5572 11.9014L20.4532 12.896C20.0062 17.1687 16.3913 20.5 11.9997 20.5C8.95904 20.5 6.28956 18.9031 4.78702 16.5H8.49974V14.5H1.49974V21.5H3.49974V18.1653C5.40769 20.7914 8.50342 22.5 11.9997 22.5C17.4261 22.5 21.8899 18.3846 22.4423 13.104L22.5464 12.1095L20.5572 11.9014Z"
              fill="currentColor"
            />
          </svg>
          <span>
            点击刷新
          </span>
        </p>
      </div>
      
      
    </div>
    
    <canvas
      role="img"
    />
    <!---->
    
  </div>
  <!-- 默认状态 -->
  <div
    class="t-qrcode"
    level="M"
    style="background-color: transparent; width: 160px; height: 160px;"
  >
    <div
      class="t-mask t-scanned"
    >
      
      <p
        class="t-scanned"
      >
        <svg
          class="t-icon t-icon-check-circle-filled t-scanned__icon"
          fill="none"
          height="1em"
          viewBox="0 0 24 24"
          width="1em"
        >
          <path
            d="M12 23C18.0751 23 23 18.0751 23 12C23 5.92487 18.0751 1 12 1C5.92487 1 1 5.92487 1 12C1 18.0751 5.92487 23 12 23ZM7.49985 10.5858L10.4999 13.5858L16.4999 7.58578L17.9141 8.99999L10.4999 16.4142L6.08564 12L7.49985 10.5858Z"
            fill="currentColor"
          />
        </svg>
        已扫描
      </p>
      
    </div>
    
    <canvas
      role="img"
    />
    <!---->
    
  </div>
</div>
`;

exports[`Qrcode > Qrcode statusVue demo works fine 1`] = `
<div
  style="display: flex; flex-direction: column;"
>
  <div
    class="t-qrcode"
    level="M"
    style="background-color: transparent; width: 160px; height: 160px;"
  >
    <!---->
    
    <canvas
      role="img"
    />
    <!---->
    
  </div>
  <p
    class="tdesign-mobile-demo-block__summary"
  >
    expired
  </p>
  <div
    class="t-qrcode"
    level="M"
    style="background-color: transparent; width: 160px; height: 160px;"
  >
    <div
      class="t-mask t-expired"
    >
      
      
      <p
        class="t-expired__text"
      >
        二维码过期
      </p>
      <p
        class="t-expired__button"
      >
        <svg
          class="t-icon t-icon-refresh"
          fill="none"
          height="1em"
          viewBox="0 0 24 24"
          width="1em"
        >
          <path
            d="M20.4997 5.83468C18.5918 3.20865 15.4961 1.5 11.9997 1.5C6.57341 1.5 2.10958 5.61542 1.55717 10.896L1.45312 11.8905L3.44227 12.0986L3.54632 11.104C3.9933 6.83125 7.60814 3.5 11.9997 3.5C15.0404 3.5 17.7099 5.09688 19.2125 7.5H15.4997V9.5H22.4997V2.5H20.4997V5.83468ZM20.5572 11.9014L20.4532 12.896C20.0062 17.1687 16.3913 20.5 11.9997 20.5C8.95904 20.5 6.28956 18.9031 4.78702 16.5H8.49974V14.5H1.49974V21.5H3.49974V18.1653C5.40769 20.7914 8.50342 22.5 11.9997 22.5C17.4261 22.5 21.8899 18.3846 22.4423 13.104L22.5464 12.1095L20.5572 11.9014Z"
            fill="currentColor"
          />
        </svg>
        点击刷新
      </p>
      
      
    </div>
    
    <canvas
      role="img"
    />
    <!---->
    
  </div>
  <p
    class="tdesign-mobile-demo-block__summary"
  >
    loading
  </p>
  <div
    class="t-qrcode"
    level="M"
    style="background-color: transparent; width: 160px; height: 160px;"
  >
    <div
      class="t-mask"
    >
      
      <div
        class="t-loading"
        style="font-size: 32px;"
      >
        <svg
          class="t-loading__gradient t-icon-loading"
          height="1em"
          style="animation-duration: 800ms; width: 32px; height: 32px;"
          version="1.1"
          viewBox="0 0 12 12"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <foreignobject
            height="12"
            width="12"
            x="0"
            y="0"
          >
            <div
              class="t-loading__gradient-conic"
            />
          </foreignobject>
        </svg>
        <!---->
        <!---->
      </div>
      
    </div>
    
    <canvas
      role="img"
    />
    <!---->
    
  </div>
  <p
    class="tdesign-mobile-demo-block__summary"
  >
    scanned
  </p>
  <div
    class="t-qrcode"
    level="M"
    style="background-color: transparent; width: 160px; height: 160px;"
  >
    <div
      class="t-mask t-scanned"
    >
      
      <p
        class="t-scanned"
      >
        <svg
          class="t-icon t-icon-check-circle-filled t-scanned__icon"
          fill="none"
          height="1em"
          viewBox="0 0 24 24"
          width="1em"
        >
          <path
            d="M12 23C18.0751 23 23 18.0751 23 12C23 5.92487 18.0751 1 12 1C5.92487 1 1 5.92487 1 12C1 18.0751 5.92487 23 12 23ZM7.49985 10.5858L10.4999 13.5858L16.4999 7.58578L17.9141 8.99999L10.4999 16.4142L6.08564 12L7.49985 10.5858Z"
            fill="currentColor"
          />
        </svg>
        已扫描
      </p>
      
    </div>
    
    <canvas
      role="img"
    />
    <!---->
    
  </div>
</div>
`;

exports[`Qrcode > Qrcode typeVue demo works fine 1`] = `
<div>
  <div
    class="t-qrcode"
    level="M"
    style="background-color: transparent; width: 160px; height: 160px;"
  >
    <!---->
    
    <canvas
      role="img"
    />
    <!---->
    
  </div>
  <div
    class="t-qrcode t-qrcode-svg"
    level="M"
    style="background-color: transparent; width: 160px; height: 160px;"
  >
    <!---->
    <svg
      height="160"
      role="img"
      viewBox="0 0 29 29"
      width="160"
    >
      <!---->
      <path
        d="M0,0 h29v29H0z"
        fill="transparent"
        shape-rendering="crispEdges"
      />
      <path
        d="M0 0h7v1H0zM8 0h1v1H8zM11 0h1v1H11zM13 0h1v1H13zM15 0h2v1H15zM18 0h2v1H18zM22,0 h7v1H22zM0 1h1v1H0zM6 1h1v1H6zM9 1h1v1H9zM11 1h1v1H11zM15 1h1v1H15zM20 1h1v1H20zM22 1h1v1H22zM28,1 h1v1H28zM0 2h1v1H0zM2 2h3v1H2zM6 2h1v1H6zM11 2h1v1H11zM13 2h1v1H13zM16 2h1v1H16zM20 2h1v1H20zM22 2h1v1H22zM24 2h3v1H24zM28,2 h1v1H28zM0 3h1v1H0zM2 3h3v1H2zM6 3h1v1H6zM9 3h2v1H9zM12 3h1v1H12zM14 3h2v1H14zM17 3h1v1H17zM19 3h1v1H19zM22 3h1v1H22zM24 3h3v1H24zM28,3 h1v1H28zM0 4h1v1H0zM2 4h3v1H2zM6 4h1v1H6zM8 4h2v1H8zM11 4h1v1H11zM13 4h5v1H13zM19 4h2v1H19zM22 4h1v1H22zM24 4h3v1H24zM28,4 h1v1H28zM0 5h1v1H0zM6 5h1v1H6zM8 5h3v1H8zM13 5h1v1H13zM16 5h2v1H16zM22 5h1v1H22zM28,5 h1v1H28zM0 6h7v1H0zM8 6h1v1H8zM10 6h1v1H10zM12 6h1v1H12zM14 6h1v1H14zM16 6h1v1H16zM18 6h1v1H18zM20 6h1v1H20zM22,6 h7v1H22zM9 7h1v1H9zM11 7h2v1H11zM14 7h1v1H14zM16 7h1v1H16zM20 7h1v1H20zM1 8h7v1H1zM9 8h1v1H9zM14 8h1v1H14zM18 8h1v1H18zM23 8h2v1H23zM28,8 h1v1H28zM1 9h2v1H1zM4 9h2v1H4zM7 9h2v1H7zM12 9h2v1H12zM15 9h1v1H15zM18 9h3v1H18zM22 9h4v1H22zM28,9 h1v1H28zM0 10h2v1H0zM3 10h5v1H3zM9 10h1v1H9zM13 10h3v1H13zM17 10h2v1H17zM20 10h3v1H20zM25 10h1v1H25zM1 11h1v1H1zM3 11h2v1H3zM8 11h1v1H8zM11 11h1v1H11zM14 11h3v1H14zM19 11h5v1H19zM25 11h1v1H25zM27 11h1v1H27zM0 12h1v1H0zM3 12h2v1H3zM6 12h1v1H6zM8 12h8v1H8zM17 12h1v1H17zM19 12h1v1H19zM21 12h1v1H21zM23 12h1v1H23zM26,12 h3v1H26zM3 13h2v1H3zM7 13h1v1H7zM11 13h4v1H11zM17 13h6v1H17zM24 13h2v1H24zM27,13 h2v1H27zM0 14h4v1H0zM6 14h2v1H6zM11 14h2v1H11zM16 14h2v1H16zM23 14h1v1H23zM5 15h1v1H5zM9 15h1v1H9zM11 15h1v1H11zM13 15h2v1H13zM17 15h2v1H17zM20 15h2v1H20zM24 15h2v1H24zM27 15h1v1H27zM1 16h1v1H1zM3 16h1v1H3zM6 16h1v1H6zM9 16h3v1H9zM14 16h3v1H14zM21 16h1v1H21zM25 16h2v1H25zM0 17h1v1H0zM2 17h4v1H2zM7 17h2v1H7zM10 17h2v1H10zM15 17h1v1H15zM17,17 h12v1H17zM0 18h1v1H0zM2 18h2v1H2zM5 18h2v1H5zM11 18h6v1H11zM18 18h1v1H18zM20 18h1v1H20zM22 18h3v1H22zM26 18h1v1H26zM0 19h1v1H0zM2 19h4v1H2zM7 19h1v1H7zM9 19h7v1H9zM19 19h1v1H19zM22 19h4v1H22zM27 19h1v1H27zM0 20h1v1H0zM3 20h1v1H3zM6 20h1v1H6zM8 20h2v1H8zM12 20h4v1H12zM17 20h1v1H17zM20 20h5v1H20zM26 20h1v1H26zM28,20 h1v1H28zM8 21h1v1H8zM11 21h3v1H11zM18 21h3v1H18zM24 21h2v1H24zM28,21 h1v1H28zM0 22h7v1H0zM8 22h3v1H8zM14 22h2v1H14zM18 22h3v1H18zM22 22h1v1H22zM24 22h1v1H24zM0 23h1v1H0zM6 23h1v1H6zM8 23h2v1H8zM11 23h1v1H11zM14 23h1v1H14zM16 23h1v1H16zM19 23h2v1H19zM24 23h2v1H24zM27,23 h2v1H27zM0 24h1v1H0zM2 24h3v1H2zM6 24h1v1H6zM8 24h2v1H8zM11 24h1v1H11zM14 24h1v1H14zM18 24h1v1H18zM20 24h7v1H20zM28,24 h1v1H28zM0 25h1v1H0zM2 25h3v1H2zM6 25h1v1H6zM8 25h1v1H8zM11 25h1v1H11zM13 25h4v1H13zM19 25h3v1H19zM23 25h1v1H23zM0 26h1v1H0zM2 26h3v1H2zM6 26h1v1H6zM8 26h1v1H8zM10 26h1v1H10zM13 26h2v1H13zM17 26h1v1H17zM19 26h9v1H19zM0 27h1v1H0zM6 27h1v1H6zM8 27h1v1H8zM12 27h2v1H12zM16 27h1v1H16zM20 27h1v1H20zM22 27h1v1H22zM24 27h2v1H24zM27 27h1v1H27zM0 28h7v1H0zM9 28h4v1H9zM17 28h1v1H17zM19 28h1v1H19zM21 28h2v1H21zM24 28h1v1H24zM26 28h1v1H26z"
        fill="#000000"
        shape-rendering="crispEdges"
      />
      <image
        height="2.9"
        width="2.9"
        x="13.05"
        y="13.05"
      />
    </svg>
  </div>
</div>
`;
